### PR TITLE
Implement EIP-7002

### PR DESCRIPTION
**Pull request by gurukamath**
### What was wrong?
EIP-7002 is not implemented in the specs

Related to Issue #939 

### How was it fixed?
Implement EIP-7002

Cute Animals - 1 of 1

### CODE REVIEW

**Review by petertdavies — DISMISSED** [review]
I have a couple of minor quibbles

File: src/ethereum/prague/fork.py [hunk 1]
```diff
@@ -505,6 +515,98 @@ class ApplyBodyOutput:
...
+        chain_id=chain_id,
+        traces=[],
+        excess_blob_gas=excess_blob_gas,
+        blob_versioned_hashes=(),
+        transient_storage=TransientStorage(),
+    )
+
+    system_tx_output = process_message_call(system_tx_message, system_tx_env)
+
+    destroy_touched_empty_accounts(
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply src/ethereum/prague/fork.py:601]: Done

File: src/ethereum/prague/blocks.py [hunk 2]
```diff
@@ -199,3 +214,41 @@ def validate_deposit_requests(
 
     if deposit_requests != expected_deposit_requests:
         raise InvalidBlock("BlockException.INVALID_REQUESTS")
+
+
+def parse_withdrawal_data(data: Bytes) -> Bytes:
```
[the remark this answers is not in the document]
Comment (gurukamath) [reply src/ethereum/prague/blocks.py]: Done

**Comment by gurukamath** [comment]
@petertdavies Have made the requested changes and additionally, removed the `json_to_requests` part of t8n tool since the tool is not required to read `requests` from the `env`. This required a minor re-factor.

File: src/ethereum/prague/blocks.py [hunk 3]
```diff
@@ -175,27 +201,53 @@ def parse_deposit_data(data: Bytes) -> Bytes:
...
+            deposit_requests += (encode_request(deposit_request),)
+
+    return deposit_requests
+
+
+def parse_withdrawal_data(data: Bytes) -> WithdrawalRequest:
+    """
+    Parses Withdrawal Request from the data.
+    """
+    assert len(data) == 76
```
Comment (SamWilsn) [inline src/ethereum/prague/blocks.py:227]: `76` is repeated a lot. Can we make a constant for it?

**Review by SamWilsn — APPROVED** [review]

**Pull request merged**